src/runtime_src/core/tools/common/tests/TestGemm.cpp

```diff
@@ -196,7 +197,12 @@ TestGemm::run(std::shared_ptr<xrt_core::device> dev)
   std::this_thread::sleep_for(std::chrono::milliseconds(30));
 
   //Calculate TOPS
-  double ipu_hclck_period= 1000000000.0/(ipu_hclock*1000000); //MHz to ns
+  if(ipu_hclock == 0) {
+    logger(ptree, "Error", "IPU H-clock is 0");
+    ptree.put("status", test_token_failed);
+    return ptree;
+  }
+  double ipu_hclck_period= 1000000000.0/((ipu_hclock)*1000000); //MHz to ns
```

Comment:
These are magic constants, are you sure you have the right number of zeros in the various places? :-)
Not overly important, just fix the formatting.

```suggestion
  double ipu_hclck_period = 1000000000.0 / (ipu_hclock * 1000000); // MHz to ns
```

Reply:
Yes! I made sure that the numbers of zeroes is right. I am not sure how else to specify them. Hence I added the comment of what this line is doing